File: app/src/main/java/ir/hosseinabbasi/payconiq/data/network/IApiService.java
package ir.hosseinabbasi.payconiq.data.network;

import java.util.List;
import java.util.Map;

import io.reactivex.Observable;
import ir.hosseinabbasi.payconiq.data.db.model.Response;
import retrofit2.http.GET;
import retrofit2.http.Path;
import retrofit2.http.QueryMap;

/**
 * Created by Dr.jacky on 2018/04/26.
 */
public interface IApiService {
    @GET("repos")
    Observable<List<Response>> getResponse(@QueryMap Map<String, String> params);
}

File: app/src/main/kotlin/ir/hosseinabbasi/payconiq/data/network/IApiService.kt
package ir.hosseinabbasi.payconiq.data.network

import io.reactivex.Observable
import ir.hosseinabbasi.payconiq.data.db.model.Response
import retrofit2.http.GET
import retrofit2.http.Path
import retrofit2.http.QueryMap

/**
 * Created by Dr.jacky on 2018/04/26.
 */
interface IApiService {
    @GET("repos")
    fun getResponse(@QueryMap params: Map<String, String>): Observable<List<Response>>
}
